vsphere/resource_vsphere_vmfs_datastore.go

```diff
@@ -0,0 +1,324 @@
+package vsphere
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+const (
+	retryDeletePending   = "retryDeletePending"
+	retryDeleteCompleted = "retryDeleteCompleted"
+	retryDeleteError     = "retryDeleteError"
+
+	waitForDeletePending   = "waitForDeletePending"
+	waitForDeleteCompleted = "waitForDeleteCompleted"
+	waitForDeleteError     = "waitForDeleteError"
+)
+
+// formatCreateRollbackError defines the verbose error for extending a disk on
+// creation where rollback is not possible.
+const formatCreateRollbackErrorUpdate = `
+WARNING: Dangling resource!
+There was an error extending your datastore with disk: %s:
+%s
+Additionally, there was an error removing the created datastore:
+%s
+You will need to remove this datastore manually before trying again.
+`
+
+// formatCreateRollbackError defines the verbose error for extending a disk on
+// creation where rollback is not possible.
+const formatCreateRollbackErrorProperties = `
+WARNING: Dangling resource!
+After creating the datastore, there was an error fetching its properties:
+%s
+Additionally, there was an error removing the created datastore:
+%s
+You will need to remove this datastore manually before trying again.
+`
+
+// formatUpdateInconsistentState defines the verbose error when the setting
+// state failed in the middle of an update opeartion. This is an error that
+// will require repair of the state before TF can continue.
+const formatUpdateInconsistentState = `
+WARNING: Inconsistent state!
+Terraform was able to add disk %s, but could not save the update to the state.
+The error was:
+%s
+This is more than likely a bug. Please report it at:
+https://github.com/terraform-providers/terraform-provider-vsphere/issues
+You will also need to repair your state before trying again.
+`
+
+func resourceVSphereVmfsDatastore() *schema.Resource {
+	s := map[string]*schema.Schema{
+		"name": &schema.Schema{
+			Type:        schema.TypeString,
+			Description: "The name of the datastore.",
+			Required:    true,
+			ForceNew:    true,
+		},
+		"host_system_id": &schema.Schema{
+			Type:        schema.TypeString,
+			Description: "The managed object ID of the host to set up the datastore on.",
+			Required:    true,
+		},
+		"disks": &schema.Schema{
+			Type:        schema.TypeList,
+			Description: "The disks to add to the datastore.",
+			Required:    true,
+			MinItems:    1,
```

Comment:
given this is required `MinItems` should be inferred?

Reply:
What I want to avoid is someone attempting to enter an empty list here - ie: `disk = []`.